### PR TITLE
Clean project to make flake8 pass

### PR DESCRIPTION
**Pull request by jorgebastida**
PEP8[1] is the de facto style guide for Python code. While releasing open source projects written in Python it is important to follow it in order to make it easier for the community to contribute.
- This PR cleans all the pep8 violations. The only exception is the `max-line-length` policy which normally is relaxed to 120 characters instead of 80.
- In order to be able to test in the future that there are no new violations, I would recommend using `flake8` against the project. In order to do so I've created a `.flake8` file which will make `flake8` understand which are the exceptions to apply (currently only the `max-line-length` one).

If in the future this project starts using Travis (or something similar) I would recommend adding `flake8` as part of the build.

[1] https://www.python.org/dev/peps/pep-0008/

### CODE REVIEW

File: tests/test_event_transformer.py [hunk 1]
```diff
@@ -375,6 +382,7 @@ def test_multiple_malformed_fields():
     except SnowplowEventTransformationException as sete:
         exception = sete
     assert(exception.error_messages == [
-        "Unexpected exception parsing field with key contexts and value bad_contexts: ValueError('No JSON object could be decoded',)",
+        ("Unexpected exception parsing field with key contexts and value "
+         "bad_contexts: ValueError('No JSON object could be decoded',)"),
```
Comment (fblundun) [inline tests/test_event_transformer.py:386]: Wow, I didn't know this syntax splitting a string across multiple lines in the source file was legal.

Comment (jorgebastida) [inline tests/test_event_transformer.py:386]: It is quite horrible... but legal, yes :)

**Comment by fblundun** [comment]
Thanks for this @jorgebastida ! I have cherry-picked this into my local master branch. When I run `sudo pip install flake8; flake8` in the repository root I still get some complaints:

```
./tests/test_event_transformer.py:349:24: F821 undefined name 'unicode'
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/event_transformer.py:17:1: F401 'json' imported but unused
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/event_transformer.py:24:1: E302 expected 2 blank lines, found 1
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/event_transformer.py:26:1: E302 expected 2 blank lines, found 0
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/event_transformer.py:28:1: E302 expected 2 blank lines, found 0
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/event_transformer.py:35:1: E302 expected 2 blank lines, found 1
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/event_transformer.py:37:1: E302 expected 2 blank lines, found 0
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/event_transformer.py:39:1: E302 expected 2 blank lines, found 0
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/event_transformer.py:41:1: E302 expected 2 blank lines, found 0
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/event_transformer.py:178:1: E302 expected 2 blank lines, found 1
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/event_transformer.py:181:1: E302 expected 2 blank lines, found 1
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/event_transformer.py:181:79: E251 unexpected spaces around keyword / parameter equals
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/event_transformer.py:181:81: E251 unexpected spaces around keyword / parameter equals
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/event_transformer.py:181:88: E261 at least two spaces before inline comment
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/event_transformer.py:183:121: E501 line too long (133 > 120 characters)
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/json_shredder.py:23:1: E302 expected 2 blank lines, found 1
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/json_shredder.py:35:1: E302 expected 2 blank lines, found 1
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/json_shredder.py:35:30: E261 at least two spaces before inline comment
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/json_shredder.py:51:1: E302 expected 2 blank lines, found 1
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/json_shredder.py:51:30: E261 at least two spaces before inline comment
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/snowplow_event_transformation_exception.py:17:1: E302 expected 2 blank lines, found 1
```

Did you see these messages when you ran flake8?

**Comment by fblundun** [comment]
@jorgebastida any update on this? Ideally we would like to release this later today.

**Comment by jorgebastida** [comment]
Sorry for replying before! Which version of python are you using? Reading the errors... it pretty much looks like the errors I fixed... have you tried doing the same in my original branch?

**Comment by fblundun** [comment]
@jorgebastida I have just tried it in your original branch using both 3.4.3 and 2.7.6 and got the same errors each time.

The error

```
./build/lib.linux-x86_64-2.7/snowplow_analytics_sdk/event_transformer.py:17:1: F401 'json' imported but unused
```

is strange because that file clearly doesn't import `json`. It makes me think that flake8 is somehow reading a different commit, or a cached copy of the file...

**Comment by fblundun** [comment]
@jorgebastida I just deleted and re-cloned my local copy of the repository and that made flake8 pass! I think it was because publishing to the PyPI test server created a .build directory with old versions of the source files, and flake8 was noticing those. Sorry for the confusion!

**Comment by fblundun** [comment]
Closing this PR since I have actually cherry-picked it into master...

**Pull request closed without merging**